src/native/minipal/log.c

```diff
@@ -239,9 +243,19 @@ static int sync_file(minipal_log_flags flags)
 
     return 0;
 }
+
+static ssize_t write_file(int fd, const char* msg, size_t bytes_to_write)
+{
+    if (fd == -1 || fd == -2)
```

Comment:
This can use a comment explaining why we check for `-2`.

Reply:
Yes, but we are discussing if we should just look for negative values and be fine with it, the reason is documented here, https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/fileno?view=msvc-170, under the Note section. So apparently, they changed the return value in case there is no console attached to -2...

Reply:
Removed the explicit checks and left it to write/_write to return error when hitting the case where there is no stdout/stderr available. According to the POSIX/Win32 documentation fileno/_fileno doesn't have a real error return and if the used FILE streams represent a file (we only use stdout/stderr as arguments), it should return something that can be used with other API's. On Windows scenario where there is no console attached to an app, _write will return -1 when called with a file descriptor representing stdout/stderr in a process without a console.